go/arrow/array/encoded.go

```diff
@@ -192,6 +193,15 @@ func (r *RunEndEncoded) GetPhysicalLength() int {
 	return encoded.GetPhysicalLength(r.data)
 }
 
+func (r *RunEndEncoded) ValueStr(i int) string {
```

Comment:
hmm. maybe I should just use then `json.Marshal(r.GetOneForMarshal(i))` for consistency in this case?

Reply:
I think that's actually the difference here between `GetOneForMarshal` and `ValueStr`. `GetOneForMarshal` would return the whole run-mapping etc. While I think `ValueStr` should just return the actual "value" for the given "logical" index, so that a consumer doesn't need to do that themselves if they are actually trying to retrieve the values from the run-end-encoded array. It also maintains consistency with the Dictionary encoded arrays, where `ValueStr` returns the actual dictionary value, not the dictionary index. Does that make sense?

We should also explicitly document this in a doc string on the function.